discojs/src/serialization/coder.ts

```diff
@@ -0,0 +1,51 @@
+import * as msgpack from "@msgpack/msgpack";
+
+export type Encoded = Uint8Array;
+
+export function isEncoded(raw: unknown): raw is Encoded {
+  if (!(raw instanceof Uint8Array)) return false;
+
+  const _: Encoded = raw;
+
+  return true;
+}
+
+// create a new buffer instead of referencing the backing one
+function copy(arr: Uint8Array): Uint8Array {
+  // `Buffer.slice` (subclass of Uint8Array on Node) doesn't copy
+  // thus doesn't respect Liskov substitution principle
+  // https://nodejs.org/api/buffer.html#bufslicestart-end
+  // here we call the correct implementation
+  return Uint8Array.prototype.slice.call(arr);
+}
+
+const CODEC = new msgpack.ExtensionCodec();
+CODEC.register({
+  type: 0x17,
```

Comment:
yes would be good to clarify in the code which models become int8 (only LLMs i guess?) or also others? because classic ML ones might not work very well with int8. LLMs can be ok

Reply:
ho right, that's quite cryptic indeed. added a few comments explaining the CODEC

the values are taken from msgpack-lite for no special reason apart that I needed random value.
Float32Array are for the weights, ArrayBuffer are for the TFJS's saved model

Reply:
to answer @martinjaggi: here it's simply the serialization we use to communicate weights & models on the network, not the types used for the tensors inside the models. theses are float32 (because WeightsContainer maps everything to float32, which should be changed at some point)
now that you mention it, if the LLM model can work with int8, we should be able to divide by four its serialized size, which is a very nice gain!

Reply:
We've been assuming that JS could only support 32 bits but can we actually train a model on int8?

Reply:
I think so, at least, there is support in TFJS for [various type of tensor](https://github.com/tensorflow/tfjs/blob/master/tfjs-core/src/types.ts#L40)